commands.js

```diff
@@ -1044,16 +1044,48 @@ exports.commands = {
 			move = Commands.randommove.call(this, "2,viable", self, dummyRoom);
 		}
 
+		//Code for generating a random name based on types generated
+		//Start by accessing the array from the capnamelist.js file in the data folder.
+		var foundName = false;
+		var t1Name = '';
+		var type1array = Data.Capnames[type[0]];
+		var type2array = Data.Capnames[type[1]];
+		while(!foundName){
+            t1Name = Tools.sample(type1array)[0]
+            if(t1Name.charAt(0) === '-'){ //Case Suffix, can't be first part of name. remove and reroll
+                Tools.removeElm(type1array, t1Name);
+            }else if(t1Name.charAt(0) === '+'){ //Case Prefix, remove '+' char from string and use
+                t1Name = t1Name.substr(1);
+                foundName = true;
+            }else{
+                foundName = true;
+            }
+		}
+		foundName = false;
+		var t2Name = '';
+		while(!foundName){ // Same stuff, but for second type
+            t2Name = Tools.sample(type2array)[0]
+            if(t2Name.charAt(0) === '+'){ //Case Prefix, can't be first part of name. remove and reroll
+                Tools.removeElm(type1array, t2Name);
+            }else if(t2Name.charAt(0) === '-'){ //Case Suffix, remove '-' char from string and use
+                t2Name = t2Name.substr(1);
+                foundName = true;
+            }else{
+                foundName = true;
+            }
+        }
+        var fullName = t1Name.charAt(0).toUpperCase() + t1Name.substr(1) + t2Name;
+        //That should be it for random name stuff - just have to add it to the output now:
 		var text = "";
 		if (!room.canHTML()) {
-			text = "``Random Fakemon:`` A **" + type.join("/") + "** Pokemon with **";
+			text = "``Random Fakemon:`` A **" + type.join("/") + "** Pokemon named "+ fullName + " with **";
```

Comment:
just replace "Random Fakemon:" with `fullName + ":...` It'll make the text flow a little better.